### ISSUE TITLE
Duplicate backend configuration  on remote.tf.json

### ISSUE THREAD

**Issue by angeloCabezas** [hashicorp/terraform-cdk#1645]
Good Morning, 

I have the next question, every time that I execute "cdktf synth" it creates a remote.tf.json, cdk.tf.json and some extra files.

Then when I tried to execute Terraform plan I get the next error : 

<img width="667" alt="image" src="https://user-images.githubusercontent.com/36689256/159846499-081b3e6f-6fc7-4204-af8d-19fd1302ca5c.png">

what it is right, because checking the remote.tf.json inside it, I have this 

<img width="659" alt="image" src="https://user-images.githubusercontent.com/36689256/159846714-7d82fec1-686e-4443-9e31-4c7ca75a3f19.png">

another terraform backend, I would like to know how to avoid this issue, because currently I'm using a automatic process which is using "cdktf synth" and after than "Terraform plan" and it fails.

Basically I dont want to create in the cdk.tf.json automatically

```
"terraform": {
    "backend": {
      "local": {
        "path": "C:\\D\\Cignium\\LM\\GitHubRepo\\infrastructure\\Platforms.LM.IAC\\src\\TerraformStack\\terraform.lm.tfstate"
      }
    },
```
Many thanks.

**Comment by jsteinich** [comment]
Are you using Terraform Cloud?
Do you have an `azurerm` backend configured somewhere?

The local backend is added automatically if no other backends are configured within the stack.

**Comment by angeloCabezas** [comment]
Hi @jsteinich 

Yes, I'm using terraform.
Yes, it was configured in two places, so I take one out.

Ok I see this last version brings backend in the same file, gotcha. 

Trying to set up terraform init, I got the following errors, looks like the way to set up is different ?? 

<img width="753" alt="image" src="https://user-images.githubusercontent.com/36689256/160028796-9728db00-486c-419f-ae80-ae172d8e8a41.png">

Could you pls walk me thought ?

**Comment by jsteinich** [comment]
Normally cdktf takes care of running `terraform init`. Your cdktf code can read in the configuration for a file, environment variable, or other sources.

If you do want to continue using terraform directly that should also be possible. It looks like you want to use Azure, so just define an instance of `AzurermBackend` within your code and put placeholders in place.

**Comment by angeloCabezas** [comment]
Hi @jsteinich 

Gotcha, many thanks for the help !

Regards

**Comment by DanielMSchmidt** [comment]
@angeloCabezas Do I read it correctly that the issue is solved?

**Comment by ansgarm** [comment]
On a related note:
We're still referring to `this.addOverride('terraform.backend'` in some places and should upgrade that to using the RemoteBackend as using an override is not detected by the logic that inserts a LocalBackend as a fallback if none was found. E.g. [our docs still recommend to use an override](https://github.com/hashicorp/terraform-cdk/blob/ac14072ba0b7e8843031d45e94af76006a4480f1/website/docs/cdktf/concepts/remote-backends.mdx#L232) but they don't mention that one would need to clear the `terraform.backend.local` via an override, too (a user figured this out [here](https://github.com/hashicorp/terraform-cdk/issues/1628#issuecomment-1074339065)).

**Comment by a0s** [comment]
I've got the duplication of all resources using the local state. I've got this behaviour two time already. I have no idea whats going on.

**Comment by ansgarm** [comment]
Hi @a0s, could you share some example code where this happens? I'm especially interested in any backend configuration and overrides. Also: are you using a monorepo and/or locally linked npm modules?
